test/suites/smoke-test-common-all/test-fee-treasury.ts

```diff
@@ -58,13 +58,17 @@ describeSuite({
                         return;
                     }
 
-                    const events = await apiAtBlock.query.system.events();
-                    for (const [_, extrinsic] of extrinsics.entries()) {
+                    const allRecords = await apiAtBlock.query.system.events();
+                    for (const [index, extrinsic] of extrinsics.entries()) {
                         // Skip unsigned extrinsics, since no commission is paid
                         if (!extrinsic.isSigned) {
                             continue;
                         }
 
+                        const events = allRecords.filter(
+                            ({ phase }) => phase.isApplyExtrinsic && phase.asApplyExtrinsic.eq(index)
```

Comment:
Nice O(n^2), can you try to split the events by index before the loop? Or too complicated?

Reply:
Sure